RELEASE.rst

```diff
@@ -0,0 +1,12 @@
+RELEASE_TYPE: patch
+
+This release improves test case reduction for recursive data structures.
+Hypothesis now guarantees that whenever a strategy calls itself recursively
```

Comment:
I would tone down this statement, to describe the new pass without guaranteeing that it will be run - just to avoid the appearance that shrinking details might be part of the public API in a compatibility sense!